src/main/java/co/foxdev/foxbotng/api/PluginManager.java

```diff
@@ -104,35 +105,39 @@ private void loadClasses() throws IOException {
         }
 
         for (Map.Entry<File, HashSet<String>> entry : jarData.entrySet()) {
-            Set<String> classNames = entry.getValue();
-            for (String className : classNames) {
+            HashSet<String> classnames = entry.getValue();
+            for (String classname : classnames) {
                 Class c;
                 try {
-                    c = this.getClass().getClassLoader().loadClass(className);
-                } catch (ClassNotFoundException e) {
+                    c = this.getClass().getClassLoader().loadClass(classname);
+                }catch(ClassNotFoundException e) {
                     continue;
                 }
 
-                if (c.isAnnotationPresent(Plugin.class)) {
+                if(c.isAnnotationPresent(Plugin.class)){
                     log.debug("Found main class: {}", c.getName());
-
-                    Object plugin;
-                    try {
-                        plugin = c.newInstance();
-                    } catch (InstantiationException | IllegalAccessException ex) {
-                        log.error("Error while loading class {}, not loading plugin.", c.getName(), ex);
-                        continue;
-                    }
-
-                    Plugin pl = plugin.getClass().getAnnotation(Plugin.class);
-                    if (plugins.containsKey(pl.name())) {
-                        log.warn("Duplicate plugin name '{}', not loading.", pl.name());
-                        return;
+                    for(Annotation annotation : c.getAnnotations()){
+                        if(annotation instanceof Plugin){
+                            Plugin pl = (Plugin) annotation;
+                            if(plugins.containsKey(pl)){
+                                log.warn("Duplicate plugin name '{}', not loading.", pl.name());
+                                break;
+                            }
+                            Object instance = null;
+                            try {
+                                instance = c.newInstance();
+                            } catch (InstantiationException | IllegalAccessException e) {
+                                log.error("Error instantiating class " + c.getName(), e);
+                            }
+                            if(instance instanceof PluginBase){
+                                log.info(String.format("Loading %s v%s (%s)", pl.name(), pl.version(), pl.author()));
```

Comment:
Don't use format(), the logger will do token replacements for you. You already do the right way on line 123.